code/modules/vehicle/atv.dm

```diff
@@ -0,0 +1,80 @@
+/obj/vehicle/atv
+	name = "all-terrain vehicle"
+	desc = "An all-terrain vehicle built for traversing rough terrain with ease. One of the few old-earth technologies that are still relevant on most planet-bound outposts."
+	icon = 'icons/vehicles/4wheeler.dmi'
+	icon_state = "fourwheel"
+	keytype = /obj/item/key
+	generic_pixel_x = 0
+	generic_pixel_y = 4
+	vehicle_move_delay = 1
+	var/static/image/atvcover = null
+
+
+/obj/vehicle/atv/New()
+	..()
+	if(!atvcover)
+		atvcover = image("icons/vehicles/4wheeler.dmi", "4wheeler_north")
+		atvcover.layer = MOB_LAYER + 0.1
+
+
+obj/vehicle/atv/post_buckle_mob(mob/living/M)
+	if(buckled_mob)
+		overlays += atvcover
+	else
+		overlays -= atvcover
+
+
+/obj/vehicle/atv/handle_vehicle_layer()
+	if(dir == SOUTH)
+		layer = MOB_LAYER+0.1
+	else
+		layer = OBJ_LAYER
+
+
+//TURRETS!
+/obj/vehicle/atv/turret
+	var/obj/machinery/gun_turret/vehicle_turret/turret = null
+
+
+/obj/machinery/gun_turret/vehicle_turret
+	name = "mounted turret"
+	scan_range = 7
+	density = 0
+
+
+/obj/vehicle/atv/turret/New()
+	..()
+	turret = new(loc)
+	//turret.base = src
```

Comment:
you guys don't have this? basically ```base``` replaces all instances of ```src``` for range checks, projectile spawning, etc. so that turrets can be "based" on another atom (such as an ATV)